src/network/message.rs

```diff
@@ -69,9 +76,6 @@ impl Encodable for CommandString {
     ) -> Result<usize, encode::Error> {
         let mut rawbytes = [0u8; 12];
         let strbytes = self.0.as_bytes();
-        if strbytes.len() > 12 {
-            return Err(encode::Error::UnrecognizedNetworkCommand(self.0.clone().into_owned()));
-        }
```

Comment:
This change will lead to a panic if the length of the CommandString is larger than 12. Why not to use `ENAMETOOLONG` `io::Error` equivalent?

Reply:
I don't think so. The PR removes the uncheck instantiation path for `CommandString` through `From` and replaces it with `CommandString::new`, which checks that the string is <=12 bytes long.